src/Common/MemoryTrackerDebugBlockerInThread.h

```diff
@@ -0,0 +1,27 @@
+#pragma once
+
+#include <base/defines.h> /// DEBUG_OR_SANITIZER_BUILD
+
+#ifdef DEBUG_OR_SANITIZER_BUILD
+#include <cstdint>
+
+struct MemoryTrackerDebugBlockerInThread
+{
+public:
+    MemoryTrackerDebugBlockerInThread();
+    ~MemoryTrackerDebugBlockerInThread();
+
+    static bool isBlocked() { return counter > 0; }
+    static constexpr bool isEnabled() { return true; }
+
+private:
+    static thread_local uint64_t counter;
```

Comment:
It is better to declare this variable in cpp file. Maybe name it `MemoryTrackerDebugBlockerInThreadCounter` wdyt?

Reply:
I  did in the same manner as other blockers done (i.e. `MemoryTrackerBlockerInThread`).

Though I like yours suggestion, since this one is different from others, because it is no-op in release builds, and it does not need "hot code path" (`isBlocked` inlined inside header) anyway (and I'm not sure that it is required for others as well, not only because of LTO, but just in general, it is called only each 4MB by default, and it access other thread local variables anyway).

Reply:
Done